packages/components/src/ComboBox/src/ComboBox.tsx

```diff
@@ -0,0 +1,337 @@
+        }
+    }, [triggerRef]);
+
+    useResizeObserver({
+        ref: triggerRef,
+        onResize: onResize
+    });
+    
+    const { stylingProps: triggerStylingProps, ...triggerOwnProps } = useStyledSystem(triggerProps ?? {});
+    const { 
+        className: triggerClassName,
+        style: triggerStyleProp,
+        ...otherTriggerProps
+    } = triggerOwnProps;
+    const { stylingProps: popoverStylingProps, ...popoverOwnProps } = useStyledSystem(popoverProps ?? {});
+    const { 
+        placement = "bottom start",
+        style: popoverStyleProp,
+        ...otherPopoverProps 
+    } = popoverOwnProps;
+    
+    const size = useResponsiveValue(sizeProp) ?? "sm";
+    const isFluid = useResponsiveValue(isFluidProp) ?? false;
+
+    const classNames = composeClassnameRenderProps(
+        className,
+        GlobalComboBoxCssSelector,
+        cssModule(
+            styles,
+            "hop-ComboBox",
+            isFluid && "fluid",
+            size
+        ),
+        stylingProps.className
+    );
+
+    const triggerClassNames = composeClassnameRenderProps(
+        triggerClassName,
+        cssModule(
+            styles,
+            "hop-ComboBox__trigger",
+            size
+        ),
+        triggerStylingProps.className
+    );
+
+    const buttonClassNames = composeClassnameRenderProps(
+        cssModule(
+            styles,
+            "hop-ComboBox__button"
+        )
+    );
+
+    const style = composeRenderProps(styleProp, prev => {
+        return {
+            ...stylingProps.style,
+            ...prev
+        };
+    });
+
+    const triggerStyle = composeRenderProps(triggerStyleProp, prev => {
+        return {
+            ...triggerStylingProps.style,
+            ...prev
+        };
+    });
+
+    const popoverStyle = composeRenderProps(popoverStyleProp, prev => {
+        return {
+            ...popoverStylingProps.style,
+            ...prev,
+            "--custom-trigger-width": triggerWidth
+        };
+    });
+    
+    const handleMouseDown: MouseEventHandler<HTMLElement> = useCallback(e => {
+        // If the input or button is the one that is clicked, we don't want to focus it since it's already done.
+        if (inputRef.current && e.target !== inputRef.current && buttonRef.current && e.target !== buttonRef.current) {
+            // forwards the focus to the input element when clicking on the input group.
+            inputRef.current.focus();
+            // This ensures that the input does not lose focus when clicking on the input group.
+            e.preventDefault();
+        }
+    }, [inputRef]);
+
+    const prefixMarkup = prefix ? (
+        <SlotProvider values={[
+            [TextContext, { size, className: styles["hop-ComboBox__prefix"] }],
+            [IconContext, { size, className: styles["hop-ComboBox__prefix"] }]
+        ]}
+        >
+            <ClearContainerSlots>
+                {isTextOnlyChildren(prefix) ? <Text>{prefix}</Text> : prefix}
```

Comment:
[Discussion]

I wonder if we should have a shared component for this, we seem to repeat this a lot.

Reply:
hmm like which part exactly? specifically isTextOnlyChildren or prefix?

Reply:
A component that would receive a children and determine to return <Text>{children}</Text> pr {children}

Reply:
hmm we certainly wouldn't do like you did for characterCount. This would be more of a utility. We probably don't need a component, it could just be a simple function. I guess components technically are just functions.. I'll figure something out, thanks.